### PR TITLE
added pthread mutex to prevent race condition in non thread safe writ…

### PR DESCRIPTION
**Pull request by tasniabari**
…e within Simulator_Log

# Quality Assurance Checklist 
To make reviews more efficient, please make sure the software feature meets the following standards and check everything off that meets the quality check. Once everything has been checked, the assigned reviewers will begin the review process. _Edit this description to check off the list._

There are exceptions with all guidelines. As long as your decisions are justified, then you are good! Contact the reviewers or the leads about any exceptions. 

## Requirements 
- [ ] Followed Coding Style Guide 
- [x] Code Build checks pass 
- [x] No merge conflicts 
- [ ] Software feature has test associated with it. 
- [ ] Software feature has documentation for it updated.
- [ ] If applicable, does software feature have simulation test associated with it (.json & out.json file)
- [ ] Test provides useful information and uses relevant data to accurately represent BPS 
    - NOTE: If test file already exists, use that one 
- [ ] 2 Members of the team have already reviewed my PR 
    - NOTE: Approver will not look at PR if 2 other members haven't reviewed first 
- [ ] Did you Have Fun?

## Things to Consider 
- [ ] Even if the above are checked, is this the best way of writing my code? 
    - It's possible to write code that works, but are there ways to make code more efficient?

### CODE REVIEW

**Review by ray4477 — APPROVED** [review]
minor change make `pthread_mutex_t SimulatorLog_Sem;` static if its only used in this file but this looks good. Im pretty confident this makes write threadsafe

**Review by connorl309 — DISMISSED** [review]
not rly any comments from me, looks good

**Comment by manthanand** [comment]
> minor change make `pthread_mutex_t SimulatorLog_Sem;` static if its only used in this file but this looks good. Im pretty confident this makes write threadsafe

This is a good comment. It is also something that should be changed (even if it would work otherwise) so next time leave it as a comment and don't approve the PR.

**Review by Champers5000 — APPROVED** [review]
meh good enough

**Review by connorl309 — APPROVED** [review]

File: BSP/Simulator/Src/Simulator.c [hunk 1]
```diff
@@ -61,7 +64,9 @@ void Simulator_Log(LoggingType_t lvl, char *str) {
     char prefix[128];
     strcpy(prefix, LoggingLUT[lvl]); //This is because strcat cannot concat const
     char* msg = strcat(prefix, str);
+    pthread_mutex_lock(&SimulatorLog_Sem);
     write(simulatorLog, msg, strlen(msg));
+    pthread_mutex_unlock(&SimulatorLog_Sem);
     printf("%s", msg);
```
[the remark this answers is not in the document]
Comment (manthanand) [reply BSP/Simulator/Src/Simulator.c:70]: printf is thread safe

**Pull request merged**